jhack/utils/simulate_event.py

```diff
@@ -126,8 +126,8 @@ def _simulate_event(
         override=env_override,
         operator_dispatch=operator_dispatch,
     )
-    # todo: insert `sudo` if this is a machine unit!
-    cmd = f"juju ssh {unit} /usr/bin/{_J_EXEC_CMD} -u {unit} {env} ./dispatch"
+    # todo: insert `sudo` if this is a machine unit! - this is now done automatically 
```

Comment:
see https://github.com/PietroPasotti/jhack/pull/40 which is already fixing the same issue. Feel free to give a review there!

IMHO it's cleaner to only use sudo when we know it's needed. On containers we don't.

Reply:
If the simulate event sticks to running from inside the machine instead of running by juju client then https://github.com/PietroPasotti/jhack/pull/40 is a greate solution to differentiate between root and standard user execution

Reply:
The latest juju documentation mentions automatic usage of sudo in case of running juju exec from client for a unit and using ubuntu user in case of application usage.

Reply:
simulate event does 'stick to running from inside the machine' as you mention. it does not run `juju exec`, instead, it runs `juju ssh <unit> /path/to/juju-exec`